temporalio/workflow.py

```diff
@@ -454,6 +478,51 @@ def now() -> datetime:
     return _Runtime.current().workflow_now()
 
 
+def patched(id: str) -> bool:
+    """Patch a workflow.
+
+    When called, this will only return true if code should take the newer path
+    which means this is either not replaying or is replaying and has seen this
+    patch before.
+
+    Use :py:func:`deprecate_patch` when all workflows are done and will never be
+    queried again. The old code path can be used at that time too.
+
+    Args:
+        id: The identifier for this patch. This identifier may be used
+            repeatedly in the same workflow to represent the same patch
+
+    Returns:
+        True if this should take the newer path, false if it should take the
+        older path.
+    """
+    return _Runtime.current().workflow_patch(id, deprecated=False)
+
+
+def random() -> Random:
+    """Get a deterministic pseudo-random number generator.
+
+    Note, this random number generator is not cryptographically safe and should
+    not be used for security purposes.
+
+    Returns:
+        The deterministically-seeded pseudo-random number generator.
+    """
+    return _Runtime.current().workflow_random()
+
+
+def uuid4() -> uuid.UUID:
+    """Get a new, determinism-safe v4 UUID based on :py:func:`random`.
+
+    Note, this UUID is not cryptographically safe and should not be used for
+    security purposes.
+
+    Returns:
+        A deterministically-seeded v4 UUID.
+    """
+    return uuid.UUID(bytes=random().getrandbits(16 * 8).to_bytes(16, "big"), version=4)
```

Comment:
This is so much better than what I had to do in TS to get random and uuid.

Reply:
Saw that. This could be even better if I was able to use `Random.randbytes()` as present in 3.9+.

I hesitated adding these APIs at all because I was hoping a sandbox could provide the `random` and `uuid` modules, but we might as well have these available now (and technically we still may offer a no-sandbox approach in the future).

Reply:
If we do have sandbox in the future we'll probably reuse these random and uuid implementations